### PR TITLE
[CDAP-7079] Add null check to timeout properties on ConfigModule

### PR DESCRIPTION
**Pull request by knazir**
Issue: https://issues.cask.co/browse/CDAP-7079
Build: http://builds.cask.co/browse/CDAP-DUT4689-1

### CODE REVIEW

**Comment by anwar6953** [comment]
Shouldn't it always be non-null?
Default values should come from cdap-default.xml

**Comment by knazir** [comment]
@anwar6953 It should, but there was an issue yesterday where this code change was pulled to an older cluster and cdap-master wasn't restarted so it was throwing a NPE. This should help guard against any issues with using outdated setting files maybe? The code path in `DefaultHTTPConfig` returns a default if the system property is not set, this just aligns with that.

@poornachandra thoughts on the comments?

**Comment by poornachandra** [comment]
Having an NPE in config module aborts the launch of container without logs, and makes it very difficult to debug issues. This check, although redundant for properly configured systems, will help us with systems that are misconfigured.

File: cdap-common/src/main/java/co/cask/cdap/common/guice/ConfigModule.java [hunk 1]
```diff
@@ -57,11 +57,16 @@ public ConfigModule(CConfiguration cConf, Configuration hConf, SConfiguration sC
...
-    System.setProperty(DefaultHttpRequestConfig.CONNECTION_TIMEOUT_PROPERTY_NAME,
-                       cConf.get(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS));
-    System.setProperty(DefaultHttpRequestConfig.READ_TIMEOUT_PROPERTY_NAME,
-                       cConf.get(Constants.HTTP_CLIENT_READ_TIMEOUT_MS));
+    // Set system properties for all HTTP requests if they were found
+    String connectionTimeout = cConf.get(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS);
+    String readTimeout = cConf.get(Constants.HTTP_CLIENT_READ_TIMEOUT_MS);
+
+    if (connectionTimeout != null) {
+      System.setProperty(DefaultHttpRequestConfig.CONNECTION_TIMEOUT_PROPERTY_NAME, connectionTimeout);
```
Comment (anwar6953) [inline cdap-common/src/main/java/co/cask/cdap/common/guice/ConfigModule.java:71]: let's add a LOG statement at least, to warn users that their system is misconfigured, somehow

**Comment by anwar6953** [comment]
LGTM

**Comment by knazir** [comment]
Guild is breen http://builds.cask.co/browse/CDAP-DUT4689-1 going to squash and merge.

**Pull request merged**